pyteal/ast/router.py

```diff
@@ -0,0 +1,547 @@
+from dataclasses import dataclass, field
+from typing import Any, cast, Optional
+from enum import Enum
+
+import algosdk.abi as sdk_abi
+
+from pyteal.config import METHOD_ARG_NUM_LIMIT
+from pyteal.errors import TealInputError
+from pyteal.types import TealType
+from pyteal.compiler.compiler import compileTeal, DEFAULT_TEAL_VERSION, OptimizeOptions
+from pyteal.ir.ops import Mode
+
+from pyteal.ast import abi
+from pyteal.ast.subroutine import (
+    OutputKwArgInfo,
+    SubroutineFnWrapper,
+    ABIReturnSubroutine,
+)
+from pyteal.ast.cond import Cond
+from pyteal.ast.expr import Expr
+from pyteal.ast.app import OnComplete, EnumInt
+from pyteal.ast.int import Int
+from pyteal.ast.seq import Seq
+from pyteal.ast.methodsig import MethodSignature
+from pyteal.ast.naryexpr import And, Or
+from pyteal.ast.txn import Txn
+from pyteal.ast.return_ import Approve
+
+
+class CallConfig(Enum):
+    """
+    CallConfigs: a "bitset" alike class for more fine-grained control over
+    `call or create` for a method about an OnComplete case.
+
+    This enumeration class allows for specifying one of the four following cases:
+    - CALL
+    - CREATE
+    - ALL
+    - NEVER
+    for a method call on one on_complete case.
+    """
+
+    NEVER = 0
+    CALL = 1
+    CREATE = 2
+    ALL = 3
+
+    def __or__(self, other: object) -> "CallConfig":
+        if not isinstance(other, CallConfig):
+            raise TealInputError("OCMethodConfig must be compared with same class")
+        return CallConfig(self.value | other.value)
+
+    def __and__(self, other: object) -> "CallConfig":
+        if not isinstance(other, CallConfig):
+            raise TealInputError("OCMethodConfig must be compared with same class")
```

Comment:
@ahangsu Minor:  There's references to _OCMethodConfig_ in several error messages.  I _think_ the referenced class name is stale.  I'm guessing it ought to read _CallConfig_.

Reply:
👍